core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1621,8 +1621,31 @@ public UnifyResult apply(UnifyRuleCall call) {
             }
           }
         }
+
+        // try to adjust RelDataType
```

Comment:
What's this change for ?
Do we have a test ?

Reply:
Can we doc the change ?

Reply:
> What's this change for ?
> Do we have a test ?

@jinxing64 Thanks for you review, this change is used for adjusting RelDataType.
Such as  `java.lang.string` and  `VARCHAR`. Left comment in jira yesterday, Maybe we can discuss it in [jira](https://issues.apache.org/jira/browse/CALCITE-3644)